src/Features/Core/Portable/ExtractClass/AbstractExtractClassRefactoringProvider.cs

```diff
@@ -67,17 +67,17 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var (document, span, cancellationToken) = context;
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var memberNodeSymbolPairs = selectedMemberNodes
-                .SelectAsArray(m => (node: m, symbol: semanticModel.GetRequiredDeclaredSymbol(m, cancellationToken)))
+                .SelectAsArray(m => (node: m, symbol: semanticModel.GetDeclaredSymbol(m, cancellationToken)))
```

Comment:
I was able to repro during the typing or modifying of a member, when a keyword is misspelled, such as `pub$$ bool DoSomething()`. Usually we wouldn't expect a user to manually trigger refactoring here but just pausing typing is enough to load possible refactorings. The node type is `IncompleteMemberSyntax`. Should we be filtering these kinds of nodes out in `NodeSelectionHelpers`, or use some other filter in each refactoring provider?

Reply:
I think it makes more sense to do the filtering in the common code path (`NodeSelectionHelpers`) to avoid duplicating code across the different refactoring providers.

Reply:
We could look to see if any of our nodes have errors (using the `HasErrors` property), and if any do, we don't perform the refactoring. I think it's a good assumption to make that error-containing code wouldn't behave nicely under refactoring.

Reply:
Could you add one test for the IncompleteMemberSyntax scenario?

Reply:
@Soreloser2 `GetDeclaredSymbol` can return null in non-error scenarios, `GlobalStatementSyntax` is an example. I don't know if this specific call is reachable with a `GlobalStatementSyntax`, but just noting that `HasErrors` *might* not be enough.

Reply:
> Should we be filtering these kinds of nodes out in NodeSelectionHelpers

Yes, that would likely be hte right thing to do.  All the callers of GetSelectedDeclarationsOrVariablesAsync should get viable declarations back.  I'd say an IncompleteMember (and/or GlobalStatementSyntax never match those).

Reply:
Alright, I'll make the change in the helpers methods to filter out `IncompleteMemberSyntax` and look into `GlobalStatementSyntax` filtering as well, and their potential equivalents for VB. There is a test to make sure we're not activating on top level statements but I'll take a closer look and write some more tests to make sure that there aren't other ways a `GlobalStatementSyntax` could be triggered.